File: frontend/src/api/deployment.ts
import request from './request'

/** 部署状态:0-部署中 1-成功 2-失败 */
export const DEPLOY_STATUS = {
  DEPLOYING: 0,
  SUCCESS: 1,
  FAILED: 2,
} as const

/** 部署请求(前端渲染好的 HTML + 简历ID + 项目名) */
export interface DeploymentRequest {
  portfolioId: number
  html: string
  projectName?: string
  description?: string
}

/** 部署记录响应 */
export interface DeploymentVO {
  id: number
  portfolioId: number
  templateId: string
  projectName: string
  description: string | null
  status: number
  deployUrl: string | null
  errorMessage: string | null
  createTime: string
  updateTime: string
}

/**
 * 部署简历模板到 EdgeOne Pages
 * <p>
 * 同步部署:后端调用 EdgeOne CLI 完成,可能耗时数十秒,
 * 此请求单独设置 200s 超时(全局默认 15s 不够)。
 */
export function deploy(data: DeploymentRequest) {
  return request.post<DeploymentVO>('/deployment', data, {
    timeout: 200000,
  }) as unknown as Promise<DeploymentVO>
}

/** 获取当前用户部署记录列表 */
export function listDeployments() {
  return request.get<DeploymentVO[]>('/deployment/list') as unknown as Promise<DeploymentVO[]>
}

/** 获取部署详情 */
export function getDeployment(id: number) {
  return request.get<DeploymentVO>(`/deployment/${id}`) as unknown as Promise<DeploymentVO>
}

File: backend/src/main/java/com/worksshow/entity/Deployment.java
package com.worksshow.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 模板部署记录实体 - 对应数据库 deployment 表
 * <p>
 * 用户将简历实例(portfolio)部署到 EdgeOne Pages 的记录。
 * 每次部署生成一条记录,关联一个简历实例。
 * 部署为同步过程,HTTP 请求阻塞等待 CLI 执行完成(通常数秒至数十秒)。
 *
 * @author WorksShow
 */
@Data
@TableName(value = "deployment", autoResultMap = true)
public class Deployment {

    /** 部署状态:0-部署中 1-成功 2-失败 */
    public static final int STATUS_DEPLOYING = 0;
    /** 部署状态:1-成功 */
    public static final int STATUS_SUCCESS = 1;
    /** 部署状态:2-失败 */
    public static final int STATUS_FAILED = 2;

    /** 主键ID(数据库自增) */
    @TableId(type = IdType.AUTO)
    private Long id;

    /** 所属用户ID(关联 user.id) */
    private Long userId;

    /** 关联简历实例ID(关联 portfolio.id) */
    private Long portfolioId;

    /** 模板ID(aurora/mo-yun/pastel,冗余便于查询) */
    private String templateId;

    /** 部署项目名称(用户自定义,展示用) */
    private String projectName;

    /** 部署描述(用户可选填写) */
    private String description;

    /** 关联自定义域名ID(关联 custom_domain.id,NULL表示用EdgeOne默认域名) */
    private Long customDomainId;

    /** 域名下子路径(同一域名区分多页面,如/portfolio1) */
    private String path;

    /** 部署状态:0-部署中 1-成功 2-失败 */
    private Integer status;

    /** 部署成功后的完整访问URL(如myresume.edgeone.app/portfolio1) */
    private String deployUrl;

    /** 部署失败时的错误信息 */
    private String errorMessage;

    /** 创建时间(部署发起时间,数据库自动维护) */
    private LocalDateTime createTime;

    /** 更新时间(部署完成时间,数据库自动维护) */
    private LocalDateTime updateTime;

    /** 逻辑删除标志:0-未删除 1-已删除(Mybatis-Plus 自动处理) */
    @TableLogic
    private Integer deleted;
}
